src/js/components/AddressBox.jsx

```diff
@@ -0,0 +1,82 @@
+import React, { Component, PropTypes } from "react";
+import { ButtonToolbar } from "react-bootstrap";
+import VoterStore from "../stores/VoterStore";
+import LoadingWheel from "../components/LoadingWheel";
+import VoterActions from "../actions/VoterActions";
+
+export default class AddressBox extends Component {
+  static propTypes = {
+    history: PropTypes.object,
+    saveUrl: PropTypes.string.isRequired
+  };
+
+  constructor (props) {
+      super(props);
+      this.state = { loading: false };
+  }
+
+  componentDidMount () {
+    this.setState({ location: VoterStore.getAddress() });
+    this.listener = VoterStore.addListener(this._onChange.bind(this));
+  }
+
+  componentWillUnmount (){
+    this.listener.remove();
+  }
+
+  _onChange () {
+    if (this.state.location){
+      this.props.history.push(this.props.saveUrl);
+    } else {
+      this.setState({ location: VoterStore.getAddress(), loading: false });
+    }
+  }
+
+  _ballotLoaded (){
+    console.log("ballotLoaded");
+    this.props.history.push(this.props.saveUrl);
+  }
+
+  updateLocation (e) {
```

Comment:
Just from my preference, I like to read code that is clear to know. We all know that e is for event, but I preference is that if you are referring to an event, name it so.